main/manager.c

```diff
@@ -6873,6 +6900,203 @@ static int action_coreshowchannels(struct mansession *s, const struct message *m
+			int size = strlen(current_channel_snapshot->base->name);
+
+			other_local = ast_malloc(size + 1);
+			if (!other_local) {
+				ast_log(LOG_ERROR, "Unable to allocate space for other local channel string\n");
+				ao2_ref(current_channel_snapshot, -1);
+				return 1;
+			}
+
+			/* Don't copy the trailing number - set it to 1 or 2, whichever one it currently is not */
+			ast_copy_string(other_local, current_channel_snapshot->base->name, size);
+			other_local[size - 1] = ast_ends_with(current_channel_snapshot->base->name, "1") ? '2' : '1';
+			other_local[size] = '\0';
+
+			other_local_snapshot = ast_channel_snapshot_get_latest_by_name(other_local);
+			if (!other_local_snapshot) {
+				ast_log(LOG_ERROR, "Unable to get other local channel\n");
+				ao2_ref(current_channel_snapshot, -1);
+				ast_free(other_local);
+				return 1;
+			}
+
+			if (coreshowchannelmap_add_to_map(channel_map, other_local_snapshot->base->name)) {
+				ast_log(LOG_ERROR, "Unable to add channel to vector\n");
+				res = 1;
+				ao2_ref(current_channel_snapshot, -1);
+				ast_free(other_local);
+				ao2_cleanup(other_local_snapshot);
+				break;
+			}
+
+			other_bridge_snapshot = ast_bridge_get_snapshot_by_uniqueid(other_local_snapshot->bridge->id);
+			if (other_bridge_snapshot) {
+				res = coreshowchannelmap_add_connected_channels(channel_map, other_local_snapshot, other_bridge_snapshot);
+			}
+
+
+			ao2_ref(current_channel_snapshot, -1);
+			ast_free(other_local);
+			ao2_cleanup(other_local_snapshot);
+			ao2_cleanup(other_bridge_snapshot);
+
+			if (res) {
+				break;
+			}
+		}
+	}
+	ao2_iterator_destroy(&iter);
+
+	return res;
+}
+
+/*! \brief  Manager command "CoreShowChannelMap" - Lists all channels connected to
+ *          the specified channel. */
+static int action_coreshowchannelmap(struct mansession *s, const struct message *m)
+{
+	const char *actionid = astman_get_header(m, "ActionID");
+	const char *channel_name = astman_get_header(m, "Channel");
+	char id_text[256];
+	int total = 0;
+	struct ast_vector_string *channel_map;
+	RAII_VAR(struct ast_bridge_snapshot *, bridge_snapshot, NULL, ao2_cleanup);
+	RAII_VAR(struct ast_channel_snapshot *, channel_snapshot, NULL, ao2_cleanup);
+
+	if (!ast_strlen_zero(actionid)) {
+		snprintf(id_text, sizeof(id_text), "ActionID: %s\r\n", actionid);
+	} else {
+		id_text[0] = '\0';
+	}
+
+	if (ast_strlen_zero(channel_name)) {
+		ast_log(LOG_ERROR, "CoreShowChannelMap requires a channel.\n");
+		return 0;
+	}
+
+	channel_snapshot = ast_channel_snapshot_get_latest_by_name(channel_name);
+	if (!channel_snapshot) {
+		astman_send_error(s, m, "Could not get channel snapshot\n");
+		return 0;
+	}
+
+	bridge_snapshot = ast_bridge_get_snapshot_by_uniqueid(channel_snapshot->bridge->id);
+	if (!bridge_snapshot) {
+		astman_send_error(s, m, "Channel is not in a bridge\n");
+		return 0;
+	}
+
+	channel_map = ao2_alloc_options(sizeof(struct ast_vector_string), channel_map_vector_destructor, AO2_ALLOC_OPT_LOCK_NOLOCK);
```

Comment:
I think instead of using an ao2_alloced vector string it would be much easier to allocate an ao2 string container:

`channel_map = ast_str_container_alloc_options(AO2_ALLOC_OPT_LOCK_NOLOCK | AO2_CONTAINER_ALLOC_OPT_DUPS_OBJ_REJECT, 1);`

This will also reject duplicates.

and then:

`ast_str_container_add(channel_map, name);`